cookbooks/aws-parallelcluster-environment/recipes/config/ebs.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+#
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+case node['cluster']['node_type']
+when 'HeadNode'
```

Comment:
Just a thought: shouldn't we have a recipe `configure_head_node` and a recipe `configure_compute_node` and put all the parts specific to a single node type there instead of checking for the node type in every recipe? I mean one entry point calling specific recipes (for instance, `ebs_head_node` vs `ebs_compute_fleet`).

Reply:
My approach is to avoid having the checks in the caller, and instead confinate the logic into the recipe itself. In this way we can always call the same set of recipes for any node_type/scheduler, then it's up to the recipe to execute the right steps according to these conditions.

I don't want to have too many files for every type of node, because in future we can have login_nodes, visualization_nodes etc, I prefer to encapsulate the logic, the steps to execute according to the node type in the recipe itself.